webapp/channels/src/components/at_mention/at_mention.tsx

```diff
@@ -32,123 +36,176 @@ type Props = {
+    const handleKeyDown = (group: Group | '') => (e: React.KeyboardEvent<HTMLAnchorElement>) => {
         if (isKeyPressed(e, Constants.KeyCodes.ENTER) || isKeyPressed(e, Constants.KeyCodes.SPACE)) {
             e.preventDefault();
 
             // Prevent propagation so that the message textbox isn't focused
             e.stopPropagation();
-            this.showOverlay(e.target as HTMLAnchorElement);
+            showOverlay(e.target as HTMLAnchorElement, group);
         }
     };
 
-    hideOverlay = () => {
-        this.setState({show: false});
+    const hideOverlay = () => {
+        setShow(false);
     };
 
-    render() {
-        const user = getUserOrGroupFromMentionName(this.props.usersByUsername, this.props.mentionName) as UserProfile | '';
+    const showGroupUserOverlay = (user: UserProfile) => {
+        hideOverlay();
+        setGroupUser(user);
+    };
 
-        if (!this.props.disableGroupHighlight && !user) {
-            const group = getUserOrGroupFromMentionName(this.props.groupsByName, this.props.mentionName) as Group | '';
+    const hideGroupUserOverlay = () => {
+        setGroupUser(undefined);
+    };
 
-            if (group && group.allow_reference) {
-                const suffix = this.props.mentionName.substring(group.name.length);
+    const returnFocus = () => {
+        document.dispatchEvent(new CustomEvent<A11yFocusEventDetail>(
+            A11yCustomEventTypes.FOCUS, {
+                detail: {
+                    target: ref.current,
+                    keyboardOnly: true,
+                },
+            },
+        ));
+    };
 
-                return (
-                    <>
-                        <span>
-                            <AtMentionGroup group={group}/>
-                        </span>
-                        {suffix}
-                    </>
-                );
-            }
+    const getPopOver = (user: UserProfile | '', group: Group | '') => {
+        if (user) {
+            return (
+                <ProfilePopover
+                    className='user-profile-popover'
+                    userId={user.id}
+                    src={Client4.getProfilePictureUrl(user.id, user.last_picture_update)}
+                    hasMention={props.hasMention}
+                    hide={hideOverlay}
+                    channelId={props.channelId}
+                />
+            );
         }
 
-        if (!user) {
-            return <React.Fragment>{this.props.children}</React.Fragment>;
+        if (group) {
+            return (
+                <UserGroupPopover
+                    group={group}
+                    hide={hideOverlay}
+                    showUserOverlay={showGroupUserOverlay}
+                    returnFocus={returnFocus}
+                />
+            );
         }
 
-        const suffix = this.props.mentionName.substring(user.username.length);
-        const displayName = displayUsername(user, this.props.teammateNameDisplay);
+        return '';
```

Comment:
should this be returning null instead ? an empty string is not a component as the two previous returns are